src/schema/schema_create.c

```diff
@@ -961,6 +977,96 @@ __create_data_source(
     return (dsrc->create(dsrc, &session->iface, uri, (WT_CONFIG_ARG *)cfg));
 }
 
+/*
+ * __create_import_cmp --
+ *     Qsort function: sort the import entries array by name.
+ */
+static int WT_CDECL
+__create_import_cmp(const void *a, const void *b)
+{
+    WT_IMPORT_ENTRY *ae, *be;
+
+    ae = (WT_IMPORT_ENTRY *)a;
+    be = (WT_IMPORT_ENTRY *)b;
+
+    return strcmp(ae->name, be->name);
+}
+
+/*
+ * __create_meta_entry_worker --
+ *     Worker function for metadata file reader procedure. The function populates the import list
+ *     with entries related to the import URI.
+ */
+static int
+__create_meta_entry_worker(WT_SESSION_IMPL *session, WT_ITEM *key, WT_ITEM *value, void *state)
+{
+    WT_IMPORT_LIST *import_list;
+    const char *meta_key, *meta_key_suffix, *meta_value;
+
+    import_list = (WT_IMPORT_LIST *)state;
+    meta_key = (const char *)key->data;
+    meta_value = (const char *)value->data;
+
+    /* Get suffix of the key. */
+    meta_key_suffix = strchr(meta_key, ':');
+    WT_ASSERT(session, meta_key_suffix != NULL && meta_key_suffix[1] != '\0');
+    ++meta_key_suffix;
+
+    /*
+     * We want to skip unrelated entries. We have stripped out the URI prefixes and want to get all
+     * the entries that match the URI. This check will match overlapping entries (i.e. if we're
+     * importing table:name but name123 also exists) but should reduce the resources needed for the
+     * list of possible entries.
+     */
+    if (!WT_PREFIX_MATCH(meta_key_suffix, import_list->uri_suffix))
```

Comment:
Hey @AlexeyAnisimovAU I was abit worried about this check here. In the case that we match overlapping entries, do we need to make sure that we do an exact string compare to check whether the string's suffix matches properly to not grab any unrelated entries?

Reply:
We discussed this offline with @jiechenbo . Everything is fine.